### PR TITLE
embind_ts: handle empty enumerators case

### PR DESCRIPTION
**Pull request by shrinktofit**
This fix in the case:

```cpp
enum class SomeEnum { a, b, c };
enum_<SomeEnum>("SomeEnum"); // No enumerators are registered.
```

the generated `.d.ts` would be something like:

```ts
// ...
type SomeEnum = // tsc: expected type here
// ...
```

after fix, the result would be:

```ts
// ...
type SomeEnum = never/* No any enumerators */;
// ...
```

### CODE REVIEW

**Review by brendandahl — CHANGES_REQUESTED** [review]
Code changes look good, but please add a test case to https://github.com/emscripten-core/emscripten/blob/main/test/other/embind_tsgen.cpp and update the reference file https://github.com/emscripten-core/emscripten/blob/main/test/other/embind_tsgen.d.ts

File: src/embind/embind_ts.js [hunk 1]
```diff
@@ -143,11 +143,15 @@ var LibraryEmbind = {
       out.push(`export interface ${this.name}Value<T extends number> {\n`);
       out.push('  value: T;\n}\n');
       out.push(`export type ${this.name} = `);
-      const outItems = [];
-      for (const [name, value] of this.items) {
-        outItems.push(`${this.name}Value<${value}>`);
+      if (this.items.length === 0) {
+        out.push('never/* No any enumerators */');
```
Comment (brendandahl) [inline src/embind/embind_ts.js:147]: `/* No any enumerators */` = `/* Empty Enumerator */`

Comment (shrinktofit) [reply src/embind/embind_ts.js]: fbe88e66a527c3a4531f0c3695ec9dc06a56b7f8

**Comment by shrinktofit** [comment]
> Code changes look good, but please add a test case to https://github.com/emscripten-core/emscripten/blob/main/test/other/embind_tsgen.cpp and update the reference file https://github.com/emscripten-core/emscripten/blob/main/test/other/embind_tsgen.d.ts

Done here: https://github.com/emscripten-core/emscripten/pull/20398/commits/c6010b25680590c6e3fff12675918460a94547ee

**Comment by shrinktofit** [comment]
I have no idea why CI failed @brendandahl

**Comment by brendandahl** [comment]
It was probably just some flakyiness in the test harness.

**Review by brendandahl — APPROVED** [review]
Thanks!

**Pull request merged**